src/containers/Noticeboard.jsx

```diff
@@ -1,125 +1,94 @@
-import React, { Component } from 'react';
-import Waypoint from 'react-waypoint';
-import { Code } from 'react-content-loader';
-import https from 'https';
-import axios from 'axios';
-import Noticecard from '../components/Noticecard'
-import '../styles/section.css';
-import '../styles/noticeboard.css';
+import React, { Component } from "react";
+import Waypoint from "react-waypoint";
+import { Code } from "react-content-loader";
+import Noticecard from "../components/Noticecard";
+import "../styles/section.css";
+import "../styles/noticeboard.css";
 
-const veritas_url = 'https://hermes.mykgp.com/';
-const notice_urls = ['acad_ug', 'acad_pg', 'bcrth', 'public'];
-
-axios.defaults.httpsAgent = new https.Agent({rejectUnauthorized: false});
+import { connect } from "react-redux";
+import { fetchNotices as fetchNotices_ } from "../actions/actions";
 
 class Noticeboard extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      notices: [],
-      notice_type: 0,
-      notice_next_page: null,
-      all_fetched: false
-    }
-    this.getNotices = this.getNotices.bind(this);
-    this.switchType = this.switchType.bind(this);
-    this._handleWaypointEnter = this._handleWaypointEnter.bind(this);
-    this._handleWaypointLeave = this._handleWaypointLeave.bind(this);
-  }
-
-  switchType(type_index) {
-    if (type_index !== this.state.notice_type){
-      this.setState({
-        notices: [],
-        notice_type: type_index,
-        notice_next_page: null,
-        all_fetched: false
-      });
-    }
-  }
-
-  getNotices(notice_type, next_page) {
-    var requestUrl = veritas_url + notice_urls[notice_type] + '/';
-    if (next_page) {
-      requestUrl = veritas_url + notice_urls[notice_type] + '/page/' + next_page;
-    }
-    axios.get(requestUrl)
-    .then(response => {
-      this.setState({
-        notices: this.state.notices.concat(response['data']['data']),
-        notice_next_page: response['data']['next_cursor']
-      });
-    }).catch(error => {
-      console.log(error);
-    })
+      noticeType: 0
+    };
+    this.props.fetchNotices(0);
   }
 
-  componentDidUpdate(prevProps, prevState) {
-    if (prevState.notice_next_page === this.state.notice_next_page && !this.state.all_fetched) {
-      this.setState({
-        all_fetched: true
-      });
-    }
-  }
+  switchType = type_index => {
+    this.setState({ noticeType: type_index });
+    // fetch first page only
+    this.props.fetchNotices(type_index, true);
+  };
 
-  _handleWaypointEnter() {
-    this.getNotices(this.state.notice_type, this.state.notice_next_page);
-  }
+  _handleWaypointEnter = () => {
```

Comment:
I'm not sure this is valid syntax inside `jsx` files. I remember eslint screaming in another project when I used this syntax. Is there a strong reason to use this over the existing one?

Reply:
I'm pretty sure arrow functions are accepted in es6 specs. The main benefit is that you don't have to do the binding in constructor (eg. this.fn = this.fn.bind(this))
Also you can keep .js extension and nothing will change

Reply:
Oh, I see. Thanks for clearing this! :smile: